src/main/java/sg/ncl/MainController.java

```diff
@@ -212,7 +212,18 @@ public String login(Model model) {
     }
     
     @RequestMapping(value="/login", method=RequestMethod.POST)
-    public String loginSubmit(@ModelAttribute("loginForm") LoginForm loginForm, Model model, HttpSession session) {
+    public String loginSubmit(
+            @Valid
+            @ModelAttribute("loginForm") LoginForm loginForm,
+            BindingResult bindingResult,
+            Model model,
+            HttpSession session) {
+
+        if (bindingResult.hasErrors()) {
+            loginForm.setErrorMsg("Login failed: Invalid email/password.");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Login failed: Invalid email/password." 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)